packages/frontend/services/web3/wagmiConnectors.tsx

```diff
@@ -27,9 +27,19 @@ export const appChains = configureChains(
       // This is ours Alchemy's default API key.
       // You can get your own at https://dashboard.alchemyapi.io
       apiKey: "oKxs-03sij-U_N0iOlrSsZFr29-IqbuF",
+      priority: 0,
     }),
-    publicProvider(),
+    publicProvider({ priority: 1 }),
   ],
+  {
+    stallTimeout: 3_000,
```

Comment:
added a `stallTimeout` and `priority` to each provider, it helps fix the issue for 429 error too many request. link to discussion : https://github.com/wagmi-dev/wagmi/discussions/1146 and solution : https://github.com/wagmi-dev/wagmi/discussions/1146#discussioncomment-4112555

Reply:
I tried experimenting with it thoroughly and here's the difference : 

To test it what I did was decrease the network speed to `slow 3G` before mainnet rpc calls are made and below are the results : 

### stallTimeout : 2_000
https://user-images.githubusercontent.com/80153681/203033490-6dcfee13-c32d-4d7e-8487-da18f0d82572.mov 

### stallTimeout : 3_000

https://user-images.githubusercontent.com/80153681/203033915-82fe172e-83de-4b65-b8e6-05bf820a3b28.mov

### Observations : 
- `wagmi` uses the second provider passed in the config array when there is no / pending response from the first RPC if the `stallTimeout` is overdue.  
- Now, since our second provider i.e mainnet publicProvider fails, the default behavior of `wagmi` is to try again the same request if it fails and thus it goes on making more requests with mainnet publiProvider (checkout first video)...

### Conclusion : 

- It seems that setting `stallTimeout : 3000` is the sweet spot because while testing for me alchemyProvider took max to max 3000ms for returning the response and thus it was not delegated to `publicProvider` but I think it might vary for others 
- The reason for not using time less than `3000ms` is sometime while testing (very rare) `alchmeyProvider` took some time for response(wihout decresing network speed) and it triggered `publicProvider` which resulted in getting 429's errors. Although it never happened when using `stallTimeout : 3000 ` . 


PS : I think `wagmi` has fixed this issue in their newer version. so in future if we upgrade it,  this will be gone more details [here](https://github.com/wagmi-dev/wagmi/discussions/1146#discussioncomment-4162077)

Reply:
Great explanation. Now I get it. Thanks :)